### ISSUE TITLE
Author Info Popup

### ISSUE THREAD

**Issue by nrshapiro** [Psychoanalytic-Electronic-Publishing/PEP-Web-User-Interface#330]
Our instructions to the keyboarders has been that if all authors share the same address, they don't need to use an ID and IDREFS for each.  This is where the app does not work correctly.  See below for an illustration, and a comparison to PEP-Web.

What should be done is 

1) Any addr without an ID should be shown at the bottom of the author info box (i).  It might be nice to provide more separation than PEP-Web currently does.
2) All author names should be shown, if there's an unassociated address (no ID)
3) If there's no unassociated address, and no nbios, you should not show an (i) icon, since it's really a dead-end and provides no information.
4) If there's no unassociated address, but there are addresses and or nbios, show the associated author names and their info 

Note also that the current system displayed data--the first author name only, doesn't even make sense in any logic.

e.g.,

![image](https://user-images.githubusercontent.com/4837554/110717275-bc799080-81d6-11eb-9ec4-49be79413307.png)

```<?xml version='1.0' encoding='UTF-8'?>
<!DOCTYPE pepkbd3 SYSTEM "http://peparchive.org/pepa1dtd/pepkbd3.dtd">
<pepkbd3><artinfo arttype="ART" j="AJP" ISSN="0002-9548" id="AJP.063.0149A">
<artyear>2003</artyear>
<artvol>63</artvol>
<artiss>2</artiss>
<artpgrg style="arabic" prefixused="">149-175</artpgrg>
<arttitle>Discriminating Among Subgroups of Borderline Personality Disorder: An Assessment of Object Representations</arttitle>
<artauth hidden="false">
<aut role="author" alias="false" listed="true" asis="false" authindexid="Tramantano, Gerald"><nfirst type="FIRST">Gerald</nfirst> <nlast>Tramantano</nlast></aut>
<aut role="author" alias="false" listed="true" asis="false" authindexid="Javier, Rafael A."><nfirst type="FIRST">Rafael</nfirst> <nmid>Art.</nmid> <nlast>Javier</nlast></aut>
<aut role="author" alias="false" listed="true" asis="false" authindexid="Colon, Mirna"><nfirst type="FIRST">Mirna</nfirst> <nlast>Colon</nlast></aut>
<autaff>
<addr>
<ln>62 Twinbrooks Trail, Chester, NJ 07930</ln>
</addr>
</autaff>
</artauth>
```

Rather than as pep-web.org does it:

![image](https://user-images.githubusercontent.com/4837554/110717238-ab308400-81d6-11eb-9a07-6b52d728a06f.png)

**Comment by bakerac4** [comment]
@nrshapiro I was re-reading https://github.com/Psychoanalytic-Electronic-Publishing/PEP-Web-User-Interface/issues/237 because there are so many different permutations of this and its fairly complex. 


What happens if you have an address with no affid ID, and multiple author roles?

**Comment by nrshapiro** [comment]
@abaker-gavant 

Isn't that covered by: 

1) Any addr without an ID should be shown at the bottom of the author info box (i). It might be nice to provide more separation than PEP-Web currently does.

It doesn't matter how many author roles there are.  It's just at the bottom of the popup.

**Comment by bakerac4** [comment]
Well, I believe each role has their own popup box. So I was confirming that the address should be in each one?

**Comment by nrshapiro** [comment]
Yes.  In the author info.  If there's no affid information, the aff applies to everyone, so if using separate boxes, it would go everywhere.  If there's an error in the keyboarded data, and there should have been affids, eventually it will be reported and fixed.

Also, reprinting the rules from the other author issue for reference here:

In the author area, group like roles together, as done in PEP Web.

1. **One popup per group, if you like, OR you present all in the same popup**...group all the name/address/bio info neatly in the popup, (The GVPi site isn't too "neat" in this, some better formatting would be welcomed.
2. If there are not affids for the authors, then there should be exactly zero or one set of affiliation data. It effectively applies to all if done that way. Or in the zero case, there is no affiliation data.
3. Section authors, via hauth, aren't defaulted to the aff data, unless it includes a matched affid.
4. For books, like the example above, I am happy to fix the main or TOC instance if it's not clear, as I have done today for the above book example. That of course depends on having proper data, as we do for many but not all books. I can't make such a blanket statement for journal articles--there are too many and we often don't get proper data for them.

**Comment by bakerac4** [comment]
@nrshapiro In an email from 1/19 you said 

```
Dear Adam,
1) Any aut with an nbio should have a popup
2) If there is an affid, the affiliation info should be included in the popup
3) If there is only one author (not counting hauth) but there is autaff data, include that at the top level (not for hauth).  If hauth has an affid, then include aff data with the hauth.
Per number 3,  for consistency they should be using the affid there but we accepted it being "defaulted" when there is one author and one affiliation.
```

Regarding item 3 from that list, do we only show autaff data with no affid id when its an a single author, or for any role with any amount of people in that role

**Comment by nrshapiro** [comment]
That's just a subcase with one author.  Also applies to multiple authors, per #2 in my previous message.

2. If there are not affids for the authors, then there should be exactly zero or one set of affiliation data. It effectively applies to all if done that way. Or in the zero case, there is no affiliation data.

ALSO: If there are no affids specified in the authors and yet there is multiple affiliation data, it's an error in the XML.  Do what you can to recover, like listing them all in each case.  Then we'll see it, see the confusion, and fix it.

**Comment by nrshapiro** [comment]
This was not fixed, at least in all cases per my testing:

Example: https://pep-web.org/browse/document/AIM.074.0441A?index=37&page=P0441

<img width="518" alt="image" src="https://user-images.githubusercontent.com/4837554/169877198-8eb09d2f-4adc-4289-877a-01983f501da9.png">

As you can see below, there's no info to show in the info icon in the XML, so no (i) should have been shown

<img width="1129" alt="image" src="https://user-images.githubusercontent.com/4837554/169877453-c01a7ca6-7646-4f93-8a22-6918df012712.png">

**Comment by nrshapiro** [comment]
More examples:

This doesn't work
- https://pep-web.org/browse/document/PDPSY.049.0404A?index=40

<img width="1131" alt="image" src="https://user-images.githubusercontent.com/4837554/169878374-b34ec078-df1c-40f4-bbea-c8857c4c087b.png">

These work:
- https://pep-web.org/browse/document/AJRPP.012.0001A
- https://pep-web.org/browse/document/ADPSA.013.0126A?page=P0126
- https://pep-web.org/browse/document/AIM.074.0023A?index=37&page=P0023

Today's version test was with production system on Chrome:

Client: 2022.04.22-522760072
Server: 8.6.3/2022.0518/v2.1.155
DataSource: v2022r1a/2022-05-19T17:25:51
Session: 4a6003c9-ad20-40fe-8a8d-6afbc80096a3

**Comment by nrshapiro** [comment]
Looking at a previous keyboardng issue regarding links, from IJP.104, I'm reminded of this bug, which seems to affect all of IJP issue 104 (at least the 4 or so articles I checked.
